app/models/user.rb

```diff
@@ -71,6 +71,13 @@ def set_next_email_as_primary
     default_email_record.update_attributes(primary: true)
   end
 
+  # Loads unread popups of a user.
+  #
+  # @return [Array<UserNotification>] The unread popup notifications
+  def unread_popups
+    notifications.popups.unread_by(self).order(created_at: :desc)
```

Comment:
this should probably be a concern. I don't think the user should know about the notification's internals.